packages/markdown-support/src/index.ts

```diff
@@ -69,10 +69,10 @@ export async function renderMarkdown(content: string, opts?: MarkdownRenderingOp
     const vfile = await parser
       .use(raw)
       .use(rehypeCollectHeaders)
-      .use(rehypeCodeBlock())
+      .use([rehypeCodeBlock()])
```

Comment:
I'm not sure about this one, I think there was a TS clash between unified's new types and [our own`Plugin` types](https://github.com/snowpackjs/astro/blob/d51e0f4fef5a19eb82c2e02c41578bd908b39720/packages/markdown-support/src/types.ts). Passing the plugin in an array is a valid `unified.use()` param and silenced the  error. Happy to take other suggestions (is someone good with unified internals and TS?)

Reply:
Could the `Pluggable` type from Unified be used https://github.com/unifiedjs/unified/blob/28475cafefb7e764ceb296090139579004f50c4a/index.d.ts#L598-L601 in place of https://github.com/snowpackjs/astro/blob/d51e0f4fef5a19eb82c2e02c41578bd908b39720/packages/markdown-support/src/types.ts#L4?

Reply:
Perhaps it’s better also to:
- type the plugin itself
- Use a normal plugin interface: a function (receiving options), which returns a transform (which receives a tree) — now `remarkCodeBlock` and `rehypeCodeBlock` are extra functions dynamically generating those?

Reply:
Unfortunately I don't think so, because our `Plugin` type is basically for an unresolved plugin before we import it in `load-plugins.ts`. `Pluggable` has types for the actual, successfully imported plugins

Reply:
Refactored the plugins as you suggested @wooorm, one less wrapping function and no more type errors 🎉